From 2af88e803f2084c17e55b835ef881b243a393fa9 Mon Sep 17 00:00:00 2001
From: Roland McGrath <roland@hack.frob.com>
Date: Tue, 8 Mar 2016 12:31:13 -0800
Subject: [PATCH] Fix tst-audit10 build when -mavx512f is not supported.

(cherry picked from commit 3bd80c0de2f8e7ca8020d37739339636d169957e)
(cherry picked from commit d603d94994a1d326ebc9e93c8be892acc834a114)
---
 sysdeps/x86_64/tst-audit10-aux.c | 6 +++---
 sysdeps/x86_64/tst-audit10.c     | 1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/sysdeps/x86_64/tst-audit10-aux.c b/sysdeps/x86_64/tst-audit10-aux.c
index 4398b8fd6f68..992a16c8adb6 100644
--- a/sysdeps/x86_64/tst-audit10-aux.c
+++ b/sysdeps/x86_64/tst-audit10-aux.c
@@ -20,13 +20,13 @@
 #include <stdlib.h>
 #include <string.h>
 
-extern __m512i audit_test (__m512i, __m512i, __m512i, __m512i,
-			   __m512i, __m512i, __m512i, __m512i);
-
 int
 tst_audit10_aux (void)
 {
 #ifdef __AVX512F__
+  extern __m512i audit_test (__m512i, __m512i, __m512i, __m512i,
+                             __m512i, __m512i, __m512i, __m512i);
+
   __m512i zmm = _mm512_setzero_si512 ();
   __m512i ret = audit_test (zmm, zmm, zmm, zmm, zmm, zmm, zmm, zmm);
 
diff --git a/sysdeps/x86_64/tst-audit10.c b/sysdeps/x86_64/tst-audit10.c
index 92e0cb4b52a5..a487b408bb0b 100644
--- a/sysdeps/x86_64/tst-audit10.c
+++ b/sysdeps/x86_64/tst-audit10.c
@@ -17,6 +17,7 @@
    <http://www.gnu.org/licenses/>.  */
 
 #include <cpuid.h>
+#include <cpu-features.h>
 
 int tst_audit10_aux (void);
 
-- 
2.10.2

